packages/nx/src/config/workspaces.ts

```diff
@@ -137,8 +141,22 @@ export class Workspaces {
       return {
         schema,
         implementationFactory,
-        batchImplementationFactory,
-        hasherFactory,
+        batchImplementationFactory: batchImplementationFactory
```

Comment:
This could be simpler if I changed the signature on `ExecutorConfig` to return a Promise. Instead I resolve lazily which keeps the signature the same